components/chat-panel.tsx

```diff
@@ -1,22 +1,25 @@
 import { useState } from 'react'
-import { type UseChatHelpers } from 'ai/react'
+import { CreateMessage, Message, type UseChatHelpers } from 'ai/react'
 import { Button } from '@/components/ui/button'
 import { PromptForm } from '@/components/prompt-form'
 import { ButtonScrollToBottom } from '@/components/button-scroll-to-bottom'
 import { CharmMenu, I_Charm, T_CharmId } from '@/components/features/prompt/prompt-charm-menu'
 import { IconRefresh, IconStop } from '@/components/ui/icons'
 import { FooterText } from '@/components/footer'
 
+type TAppend = (message: Message | CreateMessage, collectionName?: string[]) => Promise<string | null | undefined>
+
 export interface ChatPanelProps
   extends Pick<
     UseChatHelpers,
-    'append' | 'isLoading' | 'reload' | 'messages' | 'stop' | 'input' | 'setInput'
+    'isLoading' | 'reload' | 'messages' | 'stop' | 'input' | 'setInput'
   > {
   id?: string
   theme: string | undefined
+  append: TAppend
```

Comment:
We are passing an extra param "collectionNames" which changes the signature from the lib.
If we can seperate the param from this func we can put the type back the way it was.